### DISCUSSION TITLE
When compiling a package from `node_modules/` do I also need to include its dependencies explicitly?

### DISCUSSION THREAD

**Discussion by nyngwang** [babel/babel#16228] [Q&A]
For example, say I'm using `babel-loader` and trying to compile a package `A` inside `node_modules/`. Do I need to also include all (nested) dependencies of `A` explicitly for `babel-loader` to polyfill `A` correctly since all dependencies of `A` also got polyfilled?

From this answer at StackOverflow: [Webpack and Babel not transpiling a dependency inside node_modules that breaks IE 11 and Edge due to ES6's spread syntax](https://stackoverflow.com/a/60395448/5290519), it seems that we have to add them explicitly.

Do we have a better way to include all the dependencies without doing so one by one?

**Comment by Danziger** [comment]
Not sure as I haven't used Babel in a while, but you can try it yourself:

- Create a simple project with babel.
- Add the main dependency you want Babel to transpile.
- Configure your `babel.config.js` according to my StackOverflow answer you shared so that it gets transpiled.
- Navigate to one of its dependencies' source code inside `node_modules` and manually edit it to include some code that should be transpiled. For example, if the spread syntax should be transpiled, you can add a line like this:
  
      [first, ...other] = [1,2,3]; console.log(first);

You'll either see an error on the console if the non-compiled version is not supported, or "1", and then you can click the file line to see the resulting code in your bundle.
